### ISSUE TITLE
Running locally with Minikube

### ISSUE THREAD

**Issue by Chris0lsen** [mruoss/flame_k8s_backend#50]
Hi, and first of all great work on this backend! I'm trying to test it out in my local environment with minikube. I built a minimal application using the `flame_k8s_backend` and my k8s cluster is having trouble provisioning runners:

```
stream logs failed container "runner" in pod "flame-test-deployment-6664d8567c-hmtsn-8t6fz" is waiting to start: trying and failing to pull image for flame-namespace/flame-test-deployment-6664d8567c-hmtsn-8t6fz (runner)
```

I'm guessing it's trying to go to Docker hub to fetch my image, instead of looking in my minikube docker env registry? Nothing in the `k8s_client` code screams this to me, but my `flame-test` pods fire up without issue, and it's only the `runners` that fail. I feel like I'm missing something trivial :) Any advice?

The part of my `deployment.yaml` that describes my application:

```
apiVersion: apps/v1
kind: Deployment
metadata:
    name: flame-test-deployment
    namespace: flame-namespace
spec:
  replicas: 2
  selector:
    matchLabels:
      app: flame-test
  template:
    metadata:
      labels:
        app: flame-test
    spec:
      serviceAccountName: flame-test
      containers:
      - name: flame-test
        image: flame_test:latest
        ports:
        - containerPort: 4099
        - containerPort: 4369
        - containerPort: 9100
        imagePullPolicy: IfNotPresent
        env:
        - name: POD_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace 
        - name: POD_IP
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.podIP
        - name: RELEASE_DISTRIBUTION
          value: name
        - name: RELEASE_NODE
          value: flame_test@$(POD_IP)
        - name: RELEASE_COOKIE
          value: "secret"
```

Thanks in advance!

**Comment by Chris0lsen** [comment]
Maybe I can use the runner pod template model to set the imagePullPolicy to `Never` when I'm configuring the backend in my `application.ex`?

**Comment by Chris0lsen** [comment]
I uh... May have closed this one too hastily. 
```
│ [error] ** System NOT running to use fully qualified hostnames **                                           │
│ ** Hostname 10.244.0.179 is illegal **    
```

Looks like an issue when `flame` tries to `Node.connect/1` to the new pod. I've run into this in the past when using `--sname` instead of `--name` when clustering.

**Comment by mruoss** [comment]
Hey, nice to see people use it!

Regarding your last issue: check the runner pod logs. My suspicion is: the runner ood tries to delete itself but doesn't have the permissions.

Does it help to set the service account on the runner pod tpl? What permissions does it have?

```yaml
spec:
  serviceAccountName: flame-test
```

If this is the problem: leave the issue open. This shouldn't crash.

**Comment by Chris0lsen** [comment]
Thanks for taking a look! It's pretty quiet in the runner pod logs:

```
│ [notice] SIGTERM received - shutting down                                                            │
│                                                                                                      │
│ Stream closed EOF for flame-namespace/flame-test-deployment-d4ffcb948-czfn6-bxdzh (runner)  
```

And setting the SA name in the runner pod tpl didn't make a difference. I'll keep digging into this and see if I can come up with a reason.

**Comment by mruoss** [comment]
You have `min` set to 2. Besides your deployment, do you see both runner pods successfully starting?

**Comment by Chris0lsen** [comment]
I came back to this today and put my testing code in a repo so you could see where I've gone astray :) I noticed too that when I set the number of replicas to 0 (i.e. cold start), the pods start and are killed just fine - but the job never gets kicked off on the runner pods 😆 

[Here's my testing repo](https://github.com/Chris0lsen/flame_test/), I've made it as simple and dumbed-down as possible!

**Comment by mruoss** [comment]
Thanks for providing the repository.
I'm using [kind](https://kind.sigs.k8s.io/), not minikube and... works like a charm! 🙈
alright... will try minikube. Not sure what's the difference between the two...

**Comment by mruoss** [comment]
Thanks for the issue! Should be fixed with v0.5.5

**Comment by Chris0lsen** [comment]
Thanks for your help in debugging it! Great work!
